packages/core/src/components/graphs/skeleton.ts

```diff
@@ -0,0 +1,246 @@
+			this.renderGridSkeleton(showShimmerEffect);
+		} else if (this.configs.skeleton === Skeletons.VERT_OR_HORIZ) {
+			this.renderVertOrHorizSkeleton(showShimmerEffect);
+		} else if (this.configs.skeleton === Skeletons.PIE) {
+			this.renderPieSkeleton(showShimmerEffect);
+		} else if (this.configs.skeleton === Skeletons.DONUT) {
+			this.renderDonutSkeleton(showShimmerEffect);
+		} else {
+			throw new Error(`"${this.configs.skeleton}" is not a valid Skeleton type.`);
+		}
+	}
+
+	renderGridSkeleton(showShimmerEffect: boolean) {
+		this.setScales();
+		this.drawBackdrop(showShimmerEffect);
+		this.drawXGrid(showShimmerEffect);
+		this.drawYGrid(showShimmerEffect);
+		if (showShimmerEffect) {
+			this.setShimmerEffect("shimmer-lines");
+		}
+	}
+
+	renderVertOrHorizSkeleton(showShimmerEffect: boolean) {
+		const orientation = this.services.cartesianScales.getOrientation();
+		this.setScales();
+		this.drawBackdrop(showShimmerEffect);
+		if (orientation === CartesianOrientations.VERTICAL) {
+			this.drawYGrid(showShimmerEffect);
+		}
+		if (orientation === CartesianOrientations.HORIZONTAL) {
+			this.drawXGrid(showShimmerEffect);
+		}
+		this.setShimmerEffect("shimmer-lines");
+	}
+
+	renderPieSkeleton(showShimmerEffect: boolean) {
+		const outerRadius = this.computeOuterRadius();
+		const innerRadius = 0;
+		this.drawRing(outerRadius, innerRadius, showShimmerEffect);
+		if (showShimmerEffect) {
+			this.setShimmerEffect("shimmer-areas");
+		}
+	}
+
+	renderDonutSkeleton(showShimmerEffect: boolean) {
+		const outerRadius = this.computeOuterRadius();
+		const innerRadius = this.computeInnerRadius();
+		this.drawRing(outerRadius, innerRadius, showShimmerEffect);
+		if (showShimmerEffect) {
+			this.setShimmerEffect("shimmer-areas");
+		}
+	}
+
+	setScales() {
+		const xRange = this.services.cartesianScales.getMainXScale().range();
+		const yRange = this.services.cartesianScales.getMainYScale().range();
+		this.xScale = scaleLinear().domain([0, 1]).range(xRange);
+		this.yScale = scaleLinear().domain([0, 1]).range(yRange);
+	}
+
+	drawBackdrop(showShimmerEffect: boolean) {
+		const svg = this.parent;
+
+		this.backdrop = DOMUtils.appendOrSelect(svg, "svg.chart-skeleton");
+		const backdropRect = DOMUtils.appendOrSelect(this.backdrop, "rect.chart-skeleton-backdrop");
+		backdropRect
+			.attr("width", "100%")
+			.attr("height", "100%");
+
+		const [xScaleStart, xScaleEnd] = this.xScale.range();
+		const [yScaleEnd, yScaleStart] = this.yScale.range();
+
+		this.backdrop
+			.merge(backdropRect)
+			.attr("x", xScaleStart)
+			.attr("y", yScaleStart)
+			.attr("width", xScaleEnd - xScaleStart)
+			.attr("height", yScaleEnd - yScaleStart);
+
+		backdropRect
+			.classed("shimmer-effect-lines", showShimmerEffect)
+			.classed("empty-state-lines", !showShimmerEffect);
+	}
+
+	drawXGrid(showShimmerEffect: boolean) {
+		const height = this.backdrop.attr("height");
+		const width = this.backdrop.attr("width");
+		const ticksNumber = 5;
```

Comment:
do we want this in configuration so it can be changed by users? and then set 5 as the default

Reply:
![image](https://user-images.githubusercontent.com/44204353/81263015-2f5a2e00-903f-11ea-8828-089e41e2ac30.png)

I think we could reuse `numberOfTicks` from` grid`, what do you think?

Reply:
I think it's a bit much to let users customize how many ticks to show for a skeleton, IMO. :)
What about a constant in the file?